src/codegen/function_body.rs

```diff
@@ -0,0 +1,58 @@
+use std::fmt;
+
+use super::general::tabs;
+
+macro_rules! write_to_vec(
+    ($dst:expr, $($arg:tt)*) => (
+        $dst.push(fmt::format(format_args!($($arg)*)))
+    )
+);
+
+#[derive(Default, Debug)]
+pub struct Builder {
+    glib_name: String,
+    from_glib_prefix: String,
+    from_glib_suffix: String,
+    parameters: Vec<String>,
+}
+
+impl Builder {
+    pub fn new() -> Builder {
+        Default::default()
+    }
+    pub fn glib_name(mut self, name: &str) -> Builder {
```

Comment:
I don't think it matters here but FYI they [recommend against](https://aturon.github.io/ownership/builders.html#non-consuming-builders-(preferred):) consuming builders.